core/services/ardupilot_manager/main.py

```diff
@@ -31,16 +31,20 @@ def get_available_endpoints() -> Any:
     return [endpoint.asdict() for endpoint in autopilot.get_endpoints()]
 
 
-@app.post("/endpoint", status_code=status.HTTP_201_CREATED)
-def create_new_endpoint(response: Response, endpoint: Endpoint = Body(...)) -> Any:
-    if not autopilot.add_new_endpoint(endpoint):
-        response.status_code = status.HTTP_409_CONFLICT
-
-
-@app.delete("/endpoint", status_code=status.HTTP_204_NO_CONTENT)
-def remove_endpoint(response: Response, endpoint: Endpoint = Body(...)) -> Any:
-    if not autopilot.remove_endpoint(endpoint):
-        response.status_code = status.HTTP_404_NOT_FOUND
+@app.post("/endpoints", status_code=status.HTTP_201_CREATED)
+def create_endpoints(response: Response, endpoints: List[Endpoint] = Body(...)) -> Any:
+    for endpoint in endpoints:
+        if not autopilot.add_new_endpoint(endpoint):
+            response.status_code = status.HTTP_409_CONFLICT
+    autopilot.restart()
```

Comment:
will this restart if status code is 409 ?

Reply:
Yes, because status code 409 means that there was a conflict adding one or more endpoints, but not all of them, so still necessary to restart.

The other option would be to add zero or all (if any add results in conflict, add none of them, and don't restart).

Reply:
- We should at least say which one fail
- If everything fails the system will also restart
- I'm a bit confuse if "partial accept" is a ok thing to do in an API

Reply:
I'm adding the following handling:

- If there's at least one failing endpoint operation, it will report and return the failing endpoints
- If there's more than one successful endpoint, it will restart

What do you think?

Reply:
@patrickelectric we talked, and think that it should be a 100% pass/fail. If there is a problem with a single part of the configuration, we won't apply any of it.